File: LoanPaymentApp.UI/src/app/models/installment.ts
export class Installment {
  id!: number;
  code!: string;
  order: number = 0;
  startDate: Date = new Date();
  endDate: Date = new Date();
  initialDebtAmount: number = 0;
  interestAmount: number = 0;
  bsmvAmount: number = 0;
  amountToBePaid: number = 0;
  remainingInitialDebtAmount: number = 0;
  paymentPlanID!: number;
}

File: LoanPaymentApp.API/VakifBankKrediKullandirimAPI/Models/Loan/Installment.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using static VakifBankKrediKullandirimAPI.Helpers.Helpers;

namespace VakifBankKrediKullandirimAPI.Models.Loan
{
    [Index(nameof(Code), IsUnique = true)]
    public class Installment
    {
        public int Id { get; set; }
        [Required]
        public string Code { get; set; } = RandomCode();
        public int Order { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        [Precision(18, 2)]
        public decimal InitialDebtAmount { get; set; }

        [Precision(18, 2)]
        public decimal InterestAmount { get; set; }

        [Precision(18, 2)]
        public decimal BsmvAmount { get; set; }

        [Precision(18, 2)]
        public decimal AmountToBePaid { get; set; }

        [Precision(18, 2)]
        public decimal RemainingInitialDebtAmount { get; set; }

        [ForeignKey(nameof(PaymentPlan))]
        public int PaymentPlanID { get; set; }

    }
}
